include/runtime/header.h

```diff
@@ -39,6 +39,9 @@
 #define variable_block(tag) ((block *)((((uint64_t)(tag)) << 32) | 3))
 #define is_leaf_block(block) (((uintptr_t)block) & 1)
 #define is_variable_block(block) ((((uintptr_t)block) & 3) == 3)
+#define is_heap_block(s)                                                       \
+  ((((s)->h.hdr) & AGE_MASK) || (!(((s)->h.hdr) & NOT_YOUNG_OBJECT_BIT))       \
```

Comment:
the checks for AGE_MASK and FWD_PTR_BIT should be redundant, I believe.

Reply:
Great, thanks - I wasn't sure of the exact semantics of these bits. I'll correct the check to reflect this.